src/renderer/components/helpers/OpenQrUrl.tsx

```diff
@@ -80,48 +78,7 @@ export default async function processOpenQrUrl(
   const tx = window.static_translate
 
   if (url.toLowerCase().startsWith('mailto:')) {
-    log.debug('processing mailto url:', url)
-    try {
-      const accountId = selectedAccountId()
-      const mailto = parseMailto(url)
-      const messageText = mailto.subject
-        ? mailto.subject + (mailto.body ? '\n\n' + mailto.body : '')
-        : mailto.body
-
-      if (mailto.to) {
-        let contactId = await BackendRemote.rpc.lookupContactIdByAddr(
-          accountId,
-          mailto.to
-        )
-        if (contactId === null) {
-          contactId = await BackendRemote.rpc.createContact(
-            accountId,
-            mailto.to,
-            null
-          )
-        }
-        const chatId = await BackendRemote.rpc.createChatByContactId(
-          accountId,
-          contactId
-        )
-        if (messageText) {
-          await doMailtoAction(chatId, messageText)
-        } else {
-          selectChat(chatId)
-        }
-      } else {
-        if (messageText) {
-          window.__openDialog(MailtoDialog, { messageText })
-        }
-      }
-      callback && callback()
-    } catch (error) {
-      log.error('mailto decoding error', error)
-      window.__openDialog('AlertDialog', {
-        message: tx('mailto_link_could_not_be_decoded', url),
-        cb: callback,
-      })
-    }
+    processMailtoUrl(url, callback)
```

Comment:
I've moved this logic into its own method, both for reusability (it is called in two different places) and also to reduce the complexity of the `processOpenQrUrl` method